src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -0,0 +1,493 @@
+on your development machine, if needed.
+
+If your host application (`MyApp`) is already using CocoaPods, you only have to do the
+following to integrate with your `my_flutter` app:
+
+1. Add the following lines to your `Podfile`:
+
+```ruby
+  flutter_application_path = 'path/to/my_flutter/'
+  load File.join(flutter_application_path, '.ios', 'Flutter', 'podhelper.rb')
+```
+
+2. For each Xcode [target](https://guides.cocoapods.org/syntax/podfile.html#target) that needs to embed Flutter, call `install_all_flutter_pods(flutter_application_path)`.
+
+```ruby
+  target 'MyApp' do
+    install_all_flutter_pods(flutter_application_path)
+  end
+  target 'MyAppTests' do
+    install_all_flutter_pods(flutter_application_path)
+  end
+```
+
+3. Run `pod install`.
+
+Whenever you change the Flutter plugin dependencies in `some/path/my_flutter/pubspec.yaml`,
+you need to run `flutter packages get` from `some/path/my_flutter` to refresh the list
+of plugins read by the `podhelper.rb` script. Then run `pod install` again from
+`some/path/MyApp`.
+
+The `podhelper.rb` script will ensure that your plugins, the Flutter.framework, and the App.framework
+are embedded in your project.
+
+You should now be able to build the project using `⌘B`.
+
+### Under the hood
+
+If you have some reason to do this manually or debug why these steps aren't working, here's what's going on under the hood:
+
+1. `Flutter.framework` (the Engine library) is getting embedded into your app for you.  This has to match up with the release type
+(debug/profile/release) as well as the architecture for your app (arm*, i386, x86_64, etc.).  CocoaPods pulls this in as a vendored
+framework and makes sure it gets embedded into your native app.
+2. `App.framework` (your Flutter application binary) is embedded into your app.  CocoaPods also pulls this in as a vendored framework and
+makes sure it gets embedded into your native app.
+3. Any plugins are added as CocoaPod pods.  In theory, it should be possible to manually merge those in as well, but those instructions
+vary on the pod dependencies of each plugin.
+4. A build script is added to the Podfile targets that call `install_all_flutter_pods` to ensure that the binaries you build stay up to date
+with the Dart code that's actually in the folder.  It also uses your Xcode build configuration (Debug, Profile, Release) to embed the matching
+release type of `Flutter.framework` and `App.framework`.
+
+## Write code to use FlutterViewController from your host app
+
+The proper place to do this will be specific to your host app. Here is an
+example that makes sense for the blank screen of the host app generated
+by Xcode 10.0.
+
+First declare your app delegate to be a subclass of `FlutterAppDelegate`. Then define a FlutterEngine property, which help you to register a plugin without a FlutterViewController instance.
+
+In `AppDelegate.h`:
+
+```objective-c
+#import <UIKit/UIKit.h>
+#import <Flutter/Flutter.h>
```

Comment:
I changed all the `#import`s (and one include!) in the wiki yesterday to module imports or project imports.

Reply:
I changed the one include.  I did a search on github and `#import` is more popular than `@import` (~500m vs ~400m).  I think we might want to keep it with `#import`.

Reply:
I think it should be a module import using the `@import` directive. It's a relatively new directive as of clang 10, I believe, so it's not surprising it's not used much in GitHub.

`#import <Framework/Framework.h>` will automatically fall back to a module import and the module cache assuming `Framework.h` it's the umbrella header, so it's essentially the same as `@import Framework;`.  https://clang.llvm.org/docs/Modules.html#includes-as-imports

However, in `<FlutterPluginRegistrant/GeneratedPluginRegistrant.h>` `GeneratedPluginRegistrant.h` is NOT the umbrella header, so it's not equivalent.  FlutterPluginRegistrant is correctly implemented as a module.  CocoaPods generates a `FlutterPluginRegistrant-umbrella.h` but that's kind of a gross implementation detail that I wouldn't want to expose.

TL;DR we should use the suggested modern language features in our docs.

Reply:
(and there's definitely no reason to use `#include`s instead of `#imports`)

Reply:
For question on style I defer to the google objective-c style guide.  http://google.github.io/styleguide/objcguide.html#use-umbrella-headers-for-system-frameworks

It makes no recommendation between the styles, just that you use the umbrella header (implementation wise they should behave the same).  `<FlutterPluginRegistrant/GeneratedPluginRegistrant.h>` is not an umbrella import like you said, why didn't we make the umbrella header `<FlutterPluginRegistrant/FlutterPluginRegistrant.h>`?  Seems like we should have done that.  

> (and there's definitely no reason to use `#include`s instead of `#imports`)

Yep, I removed it.

Reply:
> `<FlutterPluginRegistrant/GeneratedPluginRegistrant.h>` is not an umbrella import like you said, why didn't we make the umbrella header `<FlutterPluginRegistrant/FlutterPluginRegistrant.h>`? Seems like we should have done that.

Flutter plugins were not designed with modules in mind.  ☹️  I'm working on that now. 
https://github.com/flutter/flutter/pull/41828
https://github.com/flutter/flutter/pull/42204

Apple's docs are so bad for this, but I found an indexed copy of WWDC 2014 "What's new in LLVM" that gives guidelines for imports:
https://docs.huihoo.com/apple/wwdc/2014/417_whats_new_in_llvm.pdf

> Use `@import` when importing your framework in another target
Use `#import` in the implementation to textually include the framework headers

I think one of the reasons `@import` became preferred is because it proves the framework has their module set up correctly, and therefore is importable to Swift.  `#import <Framework/Framework.h>` doesn't prove the same thing, just that that header is public in a framework since the module import is implicit.  

See my sadness trying to import our Objective-C plugins without modules as libraries here:
https://github.com/flutter/flutter/issues/41007

Reply:
Plus the Google Obj-C guide you linked to also says `@import` directives are "GOOD" but doesn't really give guidelines for when to use them, so it's not against that style guide.

Reply:
done